lib/yoti/http/request.rb

```diff
@@ -23,7 +26,19 @@ def request
     end
 
     def req
-      http_req = Net::HTTP::Get.new(uri)
+      raise RequestError, 'The request requires a HTTP method.' unless @http_method
+      raise RequestError, 'The payload needs to be a hash.' unless payload.to_s.empty? || @payload.is_a?(Hash)
+
+      case @http_method
+      when 'GET', 'DELETE'
+        http_req = Net::HTTP::Get.new(uri)
+      when 'POST', 'PUT', 'PATCH'
+        http_req = Net::HTTP::Get.new(uri)
+        http_req.set_form_data(@payload)
```

Comment:
In .NET and Java we're just adding the payload if it's present, regardless of the HTTP Method, so that it can be added to GET requests too - although based on [this SO answer ](https://stackoverflow.com/questions/978061/http-get-with-request-body)it does seem useless to add it

Reply:
@sidibos did the same for Node. I think it's unnecessary.